cpp/kiss_icp/core/VoxelHashMap.hpp

```diff
@@ -52,10 +53,14 @@ struct VoxelHashMap {
         }
     };
 
-    explicit VoxelHashMap(double voxel_size, double max_distance, int max_points_per_voxel)
+    explicit VoxelHashMap(double voxel_size,
+                          double max_distance,
+                          int max_points_per_voxel,
+                          int max_threads = tbb::task_arena::automatic)
```

Comment:
This is one of my concerns. To parameterize the voxel hash map, we would need to expose this as a variable parameter... If I'm reading this code for the first time, it would be a bit hectic I'd say, but can't figure out a better option now for controlling the number of threads

Reply:
How about migrating the correspondence searching function to `Registration`?

Quick draft:
``` c++
Vector3dVectorTuple GetCorrespondence(
    const Vector3dVector &points, 
    const VoxelHashMap &voxel_map,
    double max_correspondance_distance) const {
    ...
}
```

Interfacing iterator or const iterator in `VoxelHashMap` should make it possible to do this.